### PR TITLE
Bump up main to k8s 1.26.0

### LINKED ISSUE #2477: Bump up main to Kubernetes 1.26.0-RC

### PR DESCRIPTION
**Pull request by jnummelin**
Signed-off-by: Jussi Nummelin <jnummelin@mirantis.com>

## Description

Some usual go.mod shenanigangs needed to be sorted out.

There is couple breaking Golang API changes which needed to be fixed.

Fixes #2477

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

- [ ] Manual test
- [x] Auto test added

## Checklist:

- [x] My code follows the style [guidelines](https://github.com/k0sproject/k0s/blob/main/docs/contributors/overview.md) of this project 
- [x] My commit messages are [signed-off](https://github.com/k0sproject/k0s/blob/main/docs/contributors/github_workflow.md)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by jnummelin** [comment]
Created still as draft to see how smokes behave. Did run couple of them locally, but let's see how CI sees them...

**Comment by jnummelin** [comment]
BYO CRI test is a real failure. Boils down to kubelet only now supporting v1 of CRI API where as cri-dockerd only supports the v1alpha API version.
```
time="2022-12-08 06:01:40" level=info msg="E1208 06:01:39.997672     833 run.go:74] \"command failed\" err=\"failed to run Kubelet: validate service connection: CRI v1 runtime API is not imple
mented for endpoint \\\"unix:///var/run/cri-dockerd.sock\\\": rpc error: code = Unimplemented desc = unknown service runtime.v1.RuntimeService\"" component=kubelet
```
ref: https://github.com/Mirantis/cri-dockerd/issues/125

**Comment by jnummelin** [comment]
Some of the kubelet CSR related timings seem to have changed and `check-basic` smoke seem to be bit flaky now:
```
    basic_test.go:82: 
                Error Trace:    /home/jussi/go/src/github.com/k0sproject/k0s/inttest/basic/basic_test.go:82
                Error:          Received unexpected error:
                                no CSRs have been approved
                Test:           TestBasicSuite/TestK0sGetsUp
```

Looking at the test code I'm wondering how this has not been flaky in the past actually. 😄 I think we need to actually poll the CSR statuses to see them get approved rather than have it only as one-shot test.

File: pkg/autopilot/controller/readyprober.go [hunk 1]
```diff
@@ -114,15 +114,19 @@ func (p readyProber) probeOne(target apv1beta2.PlanCommandTargetStatus) error {
 		return fmt.Errorf("no internal IP address found for %v", target.Name)
 	}
 
-	url, err := url.Parse(fmt.Sprintf(readyzURLFormat, address))
-	if err != nil {
-		return fmt.Errorf("unable to parse URL for '%s': %w", address, err)
-	}
+	// url, err := url.Parse(fmt.Sprintf(readyzURLFormat, address))
```
Comment (soider) [inline pkg/autopilot/controller/readyprober.go:117]: leftover

**Review by soider — DISMISSED** [review]
approved with minor leftover to clean

File: pkg/autopilot/controller/readyprober.go (hunk 1 above)
Comment (jnummelin) [reply pkg/autopilot/controller/readyprober.go]: Fixed. Require re-stamping though

**Review by makhov — APPROVED** [review]

**Pull request merged**